libraries/ESP8266WebServer/examples/Graph/Graph.ino

```diff
@@ -0,0 +1,334 @@
+  // FILESYSTEM INIT
+
+  fileSystemConfig.setAutoFormat(false);
+  fileSystem->setConfig(fileSystemConfig);
+  boolean fsOK = fileSystem->begin();
+  DBG_OUTPUT_PORT.println(fsOK ? F("Filesystem initialized.") : F("Filesystem init failed!"));
+
+  ////////////////////////////////
+  // PIN INIT
+  pinMode(4, INPUT);
+  pinMode(12, OUTPUT);
+  pinMode(13, OUTPUT);
+  pinMode(15, OUTPUT);
+
+  ////////////////////////////////
+  // WI-FI INIT
+  DBG_OUTPUT_PORT.printf("Connecting to %s\n", ssid);
+  WiFi.mode(WIFI_STA);
+  WiFi.begin(ssid, password);
+  // Wait for connection
+  while (WiFi.status() != WL_CONNECTED) {
+    delay(500);
+    DBG_OUTPUT_PORT.print(".");
+  }
+  DBG_OUTPUT_PORT.println("");
+  DBG_OUTPUT_PORT.print(F("Connected! IP address: "));
+  DBG_OUTPUT_PORT.println(WiFi.localIP());
+
+  ////////////////////////////////
+  // MDNS INIT
+  if (MDNS.begin(host)) {
+    MDNS.addService("http", "tcp", 80);
+    DBG_OUTPUT_PORT.print(F("Open http://"));
+    DBG_OUTPUT_PORT.print(host);
+    DBG_OUTPUT_PORT.println(F(".local to open the graph page"));
+  }
+
+  ////////////////////////////////
+  // WEB SERVER INIT
+
+  //get heap status, analog input value and all GPIO statuses in one json call
+  server.on("/espData", HTTP_GET, []() {
+    String json;
+    json.reserve(88);
+    json = "{\"time\":";
+    json += millis();
+    json += ", \"heap\":";
+    json += ESP.getFreeHeap();
+    json += ", \"analog\":";
+    json += analogRead(A0);
+    json += ", \"gpioMask\":";
+    json += gpioMask;
+    json += ", \"gpioData\":";
+    json += (uint32_t)(((GPI | GPO) & 0xFFFF) | ((GP16I & 0x01) << 16));
+    json += "}";
+    server.send(200, "text/json", json);
+  });
+
+  // Default handler for all URIs not defined above
+  // Use it to read files from filesystem
+  server.onNotFound(handleNotFound);
+
+
+  // Start server
+  server.begin();
+  DBG_OUTPUT_PORT.println("HTTP server started");
+
+  DBG_OUTPUT_PORT.println("Please pull GPIO4 low (e.g. press button) to switch output mode:");
+  DBG_OUTPUT_PORT.println(" 0 (OFF):    outputs are off and hidden from chart");
+  DBG_OUTPUT_PORT.println(" 1 (AUTO):   outputs are rotated automatically every second");
+  DBG_OUTPUT_PORT.println(" 2 (MANUAL): outputs can be toggled from the web page");
+
+}
+
+// Return default GPIO mask, that is all I/Os except SD card ones
+unsigned int defaultMask() {
+  unsigned int mask = 0b11111111111111111;
+  for (auto pin = 0; pin <= 16; pin++) {
+    if (isFlashInterfacePin(pin)) {
+      mask &= ~(1 << pin);
+    }
+  }
+  return mask;
+}
+
+int rgbMode = 1; // 0=off - 1=auto - 2=manual
+int rgbValue = 0;
+long lastChangeTime = 0;
+boolean modeChangeRequested = false;
+
+void loop(void) {
+  server.handleClient();
+  MDNS.update();
+
+  if (digitalRead(4) == 0) {
+    // button pressed
+    modeChangeRequested = true;
+  }
+
+  // see if one second has passed since last change
+  long now = millis();
+  if (now - lastChangeTime > 1000) {
```

Comment:
Although tbis ks technically correct, it reinvents the wheel. Please use a global esp8266::polledTimeout::periodicMs, see core examples for use.

Reply:
Also, prefer early return to reduce cyclomatic complexity of what follows, i. e. 
```cpp
esp8266::polledTimeout::periodicMs timeToChange(1000);
...

if(!timeToChange)
    return;

if(modeChangeRequested) {
... 
```

Reply:
Didn't know about periodicMs. Nice :-)